enterprise/management/commands/update_role_assignments_with_customers.py

```diff
@@ -0,0 +1,338 @@
+            dest='enterprise_customer_uuid',
+            help='The enterprise customer to limit role assignments to.',
+        )
+        parser.add_argument(
+            '--dry-run',
+            action='store_true',
+            dest='dry_run',
+            help=(
+                'If set, no updates or creates will occur; will instead iterate over '
+                'the assignments that would be modified or created'
+            ),
+        )
+        parser.add_argument(
+            '--undo',
+            action='store_true',
+            dest='undo',
+        )
+        parser.add_argument(
+            '--args-from-database',
+            action='store_true',
+            dest='args_from_database',
+            help='If true, read arguments from a DB config model.',
+        )
+
+    def handle(self, *args, **options):
+        """
+        Entry point for managment command execution.
+        """
+        valid_roles = (ENTERPRISE_ADMIN_ROLE, ENTERPRISE_LEARNER_ROLE, ENTERPRISE_OPERATOR_ROLE)
+
+        if options['args_from_database']:
+            options.update(self._latest_settings_from_db())
+
+        if options['undo']:
+            self._undo()
+            return
+
+        if options['role'] is None:
+            for role in valid_roles:
+                self._handle(
+                    options['batch_size'],
+                    role,
+                    enterprise_customer_uuid=options.get('enterprise_customer_uuid', None),
+                    dry_run=options['dry_run']
+                )
+        elif options['role'] in valid_roles:
+            self._handle(
+                options['batch_size'],
+                options['role'],
+                enterprise_customer_uuid=options.get('enterprise_customer_uuid', None),
+                dry_run=options['dry_run'],
+            )
+        else:
+            log.error(f"{options['role']} is not a valid role.")
+            return
+
+    def _undo(self):
+        """
+        Helper that resets the ``enterprise_customer`` and ``applies_to_all_contexts``
+        fields to null/false respectively on every enterprise role assignment instance.
+        Should likely not be used in a production environment.
+        TODO: delete me.
+        """
+        for assignment in SystemWideEnterpriseUserRoleAssignment.objects.all():
+            assignment.enterprise_customer = None
+            assignment.applies_to_all_contexts = False
+            assignment.save()
+
+    def _handle(self, batch_size, role_name, enterprise_customer_uuid=None, dry_run=False):
+        """
+        Does all of the actual work on assignments of a given role.
+        """
+        if role_name == ENTERPRISE_OPERATOR_ROLE:
+            self._handle_operators(dry_run)
+            return
+
+        ent_customer_user_ids = self._get_all_enterprise_customer_user_ids(enterprise_customer_uuid)
+
+        log.info(f'There are {len(ent_customer_user_ids)} total enterprise customer user ids to process.')
```

Comment:
why not change this string to also be 'string'.format so the pylint error doesn't need to be disabled?

Reply:
The newer version of edx-lint actually allows f-strings, so I'll just delete the pylint pragma.

Reply:
...but maybe we need to get on that new version first?  I'm putting the disable back in for now.